lib/src/_shared/PickerPopper.tsx

```diff
@@ -0,0 +1,143 @@
+import Popper, { PopperProps } from '@material-ui/core/Popper';
+import TrapFocus, { TrapFocusProps } from '@material-ui/core/Unstable_TrapFocus';
+import { useForkRef } from '@material-ui/core/utils';
+import { makeStyles } from '@material-ui/core/styles';
+import { useGlobalKeyDown, keycode } from './hooks/useKeyDown';
+import { IS_TOUCH_DEVICE_MEDIA } from '../constants/dimensions';
+import { executeInTheNextEventLoopTick } from '../_helpers/utils';
+import { TransitionProps } from '@material-ui/core/transitions/transition';
+
+export interface ExportedPickerPopperProps {
+  /**
+   * Popper props passed down to material-ui [Popper](https://material-ui.com/api/popper/#popper-api) component.
+   */
+  PopperProps?: Partial<PopperProps>;
+  /**
+   * Custom component for [Transition](https://material-ui.com/components/transitions/#transitioncomponent-prop).
+   */
+  TransitionComponent?: React.ComponentType<TransitionProps>;
+}
+
+export interface PickerPopperProps extends ExportedPickerPopperProps, PaperProps {
+  role: 'tooltip' | 'dialog';
+  TrapFocusProps?: Partial<TrapFocusProps>;
+  anchorEl: PopperProps['anchorEl'];
+  open: PopperProps['open'];
+  onClose: () => void;
+  onOpen: () => void;
+}
+
+export const useStyles = makeStyles(
+  theme => ({
+    popper: {
+      zIndex: theme.zIndex.modal,
+    },
+    paper: {
+      transformOrigin: 'top center',
+      '&:focus': {
+        outline: 'auto',
+        [IS_TOUCH_DEVICE_MEDIA]: {
+          outline: 0,
+        },
+      },
+    },
+    topTransition: {
+      transformOrigin: 'bottom center',
+    },
+  }),
+  { name: 'MuiPickersPopper' }
+);
+
+export const PickerPopper: React.FC<PickerPopperProps> = ({
+  role,
+  PopperProps,
+  TrapFocusProps,
+  TransitionComponent = Grow,
+  open,
+  innerRef = null,
+  anchorEl,
+  children,
+  onClose,
+  onOpen,
+  ...other
+}) => {
+  const classes = useStyles();
+  const lastFocusedElementRef = React.useRef<Element | null>(null);
+  const paperRef = React.useRef<HTMLElement>(null);
+  const handlePopperRef = useForkRef(paperRef, innerRef);
+
+  useGlobalKeyDown(open, {
+    [keycode.Esc]: onClose,
+  });
+
+  React.useEffect(() => {
+    if (role === 'tooltip') {
+      return;
+    }
+
+    if (open) {
+      lastFocusedElementRef.current = document.activeElement;
+    } else if (
+      lastFocusedElementRef.current &&
+      lastFocusedElementRef.current instanceof HTMLElement
+    ) {
+      lastFocusedElementRef.current.focus();
+    }
+  }, [open, role]);
+
+  const handleBlur = () => {
+    if (!open) {
+      return;
+    }
+
+    // document.activeElement is updating on the next tick after `blur` called
+    executeInTheNextEventLoopTick(() => {
+      if (paperRef.current?.contains(document.activeElement)) {
+        return;
+      }
+
+      onClose();
+    });
+  };
+
+  return (
+    <Popper
+      transition
+      role={role}
+      open={open}
+      anchorEl={anchorEl}
+      placement="bottom"
```

Comment:
```suggestion
      placement="bottom-start"
```

per #1648

Reply:
Looks awful as for me, maybe we need to provide shorthand prop for this case? To prevent using `PopperProps={{ placement: "bottom-start" }}`

![image](https://user-images.githubusercontent.com/16926049/85698599-7e3d4f00-b6e3-11ea-9f49-ff2dcc12ceeb.png)

Reply:
`placement="bottom-start"` should be the default, however, I think that developers should also be able to customize the default placement. For instance, we have seen people asking for `placement="bottom-end"`

Reply:
Where says that it should be default? It looks askew on the page
![image](https://user-images.githubusercontent.com/16926049/85699984-a37e8d00-b6e4-11ea-8fe3-1ad928120642.png)

Reply:
Yes, exactly, now, it also lack two other important changes:

1. The reference point is wrong: #1768 (wrong ref)
2. The spacing of the desktop is wrong #1648.

Once done, it should be 👌